### ISSUE TITLE
mount -freespace emulation needs to report free space relative to the amount given

### ISSUE THREAD

**Issue by joncampbell123** [joncampbell123/dosbox-x#1894]
**Is your feature request related to a problem? Please describe.**
The -freespace emulation currently reports the specified amount of freespace regardless of folder contents.

**Describe the solution you'd like**
Free space emulation needs to report free space relative to the amount given. If you mount a folder with -freespace 16, it should report 16MB, and if you copy 4MB of data onto that drive, it should report 16MB - 4MB  = 12MB of free space.

For the time being it doesn't have to be 100% accurate. It just has to reduce the reported amount on file writes (that extend the EOF) and it has to increase the reported amount (up to the cap) when files are truncated or deleted.

![Firefox_Screenshot_2020-09-30T22-31-37 340Z](https://user-images.githubusercontent.com/6245486/94746646-0b6ef000-0332-11eb-8f57-72f36147f914.png)

**Comment by Wengier** [comment]
I guess the existing "freesizecap" config option can be modified to add support for this dynamic reporting as well - when enabled DOSBox-X will record the free disk size when the drive was mounted, and calculate the difference between that free disk size and the current free disk size, then add this difference to the size specified by the -freesize option. I think the result will likely be pretty accurate.

**Comment by Wengier** [comment]
In general the more accurate, the better. If you are fine with the above I can implement it quickly for the next version.

**Comment by joncampbell123** [comment]
It's getting close to build night.

I got VS2019 installed and working again, I'll start building tonight.

**Comment by Wengier** [comment]
@joncampbell123 I see. In any case I will implement it shortly after the current release so that there is more time for testing to be free of possible bugs.

**Comment by joncampbell123** [comment]
In unrelated news, if you noticed dosbox-x.software was stuck on a domain parking page for a day, it's because I finally got Network Solutions to transfer the domain to Epik, but I had to update the domain name servers and A/AAAA records behind it to get it working again.

**Comment by Wengier** [comment]
@joncampbell123 So that was why it did not work when I tried it yesterday. I did think it was likely dead. But dosbox-x.com is always shorter and easier to type of course.

**Comment by Wengier** [comment]
@joncampbell123 http://dosbox-x.software does work now. Do you want dosbox-x.software to be still mentioned somewhere in documentation? Just in case you prefer to see it as well.

**Comment by joncampbell123** [comment]
Certainly, let's have the documentation mention both dosbox-x.com or dosbox-x.software.

**Comment by Wengier** [comment]
@joncampbell123 Sure! I will do it right now!

**Comment by Wengier** [comment]
Both domains are now shown in the repository documentation (README, installer doc, etc), project Wiki and homepage itself. In certain other files the URL seems to be a quoted string and can contain only contain one URL, and thus I had not changed these ones.

**Comment by MobyGamer** [comment]
> In general the more accurate, the better. If you are fine with the above I can implement it quickly for the next version.

Appreciated (I'm the requestor).  The use case for this is testing cache management code for https://github.com/MobyGamer/total-dos-launcher , where a full cache will then be managed using various strategies.

**Comment by Wengier** [comment]
@MobyGamer I have already enhanced the config option "freesizecap" to allow the new setting "dynamic" (i.e. "freesizecap=dynamic"), which when set will report the free disk space relative to the amount given. Turns out it required slightly more work than I originally thought, but it seems to work here (both Windows and Linux). Thanks for the suggestion!

**Comment by Wengier** [comment]
I think the result should be considered reliable during my tests, but please note that it will include free size changes caused by outside environments too, not just changes inside DOSBox-X.

**Comment by MobyGamer** [comment]
That's exactly how I'd want it implemented, honestly, since an outside mount is an outside mount.  If people want it to remain fixed, they can simply not use freesizecap=relative.

Thanks so much!  I'll wait for the next Windows build and then test it.
